graphql/database/users/getUserSearchEngine.js

```diff
@@ -4,6 +4,65 @@ import getUserFeature from '../experiments/getUserFeature'
 import getSearchEngine from '../search/getSearchEngine'
 import { DatabaseItemDoesNotExistException } from '../../utils/exceptions'
 import getWidgets from '../widgets/getWidgets'
+import ReferralDataModel from '../referrals/ReferralDataModel'
+import logger from '../../utils/logger'
+
+const searchEnginesToAddDimensions = ['SearchForACause', 'Yahoo']
+
+/**
+ * Customize the search engine data to the user, as needed.
+ * @param {object} userContext - The user authorizer object.
+ * @param {string} user - UserModel object.
+ * @param {string} searchEngineId - The ID of the user's search engin
+ * @return {SearchEngine}  A SearchEngineModel instance representing the engine displayed to the user.
+ */
+const generateSearchEngine = async (userContext, user, searchEngineId) => {
+  const engineData = getSearchEngine(searchEngineId)
+  const { searchUrl } = engineData
+
+  let searchUrlPersonalized = searchUrl
+
+  // For SFAC & Yahoo, modify the personalized search URL to include
+  // dimensions used in aggregated analytics (e.g., search revenue by cause).
+  try {
+    if (searchEnginesToAddDimensions.indexOf(engineData.id) > -1) {
+      const { causeId, v4BetaEnabled, id: userId } = user
+      let referringChannel
+      try {
+        ;({ referringChannel } = await ReferralDataModel.get(
```

Comment:
What is the purpose of the semicolon here?

Reply:
@jedtan JS does automatic semicolon insertion ([light reading](https://262.ecma-international.org/#sec-automatic-semicolon-insertion)), so they're mostly not necessary, but there are a few situations where omitting them is dangerous: mostly for lines that begin with `[` or `(`. Prettier describes [this here](https://prettier.io/docs/en/rationale.html#semicolons).